lego/apps/events/views.py

```diff
@@ -271,6 +274,51 @@ def upcoming(self, request):
         serializer = self.get_serializer(queryset, many=True)
         return Response(serializer.data)
 
+    @decorators.action(
+        detail=True, methods=["GET"], serializer_class=EventAdministrateSerializer
+    )
+    def export(self, *args, **kwargs):
+        if not self.request.user.is_authenticated:
+            return Response(status=status.HTTP_401_UNAUTHORIZED)
+
+        has_permission = self.request.user.has_perm("administrate", obj=Event)
+        if not has_permission:
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        query = (
+            Event.objects.filter(pk=kwargs["pk"])
+            .prefetch_related(
+                Prefetch(
+                    "pools__registrations",
+                    queryset=Registration.objects.select_related(
+                        "user"
+                    ).prefetch_related("user__abakus_groups"),
+                ),
+            )
+            .first()
+        )
+
+        event = self.get_serializer(query).data
+
+        response = HttpResponse(content_type="text/csv")
+        response[
+            "Content-Disposition"
+        ] = f'attachment; filename="{event["title"].replace(" ", "_")}_registrations.csv"'
+        writer = csv.writer(response)
+        writer.writerow(["username", "fullName", "emailAddress", "pool"])
+        for pool in event["pools"]:
+            for registration in pool["registrations"]:
+                writer.writerow(
+                    [
+                        registration["user"]["username"],
```

Comment:
A lot of times they want eg. allergies as well, but they are not required every time. Feedback is also nice for xKOM, but not useful or necessary when giving a list to the venue owner. Same for email. Less is more

Reply:
I agree. I think it's very difficult to know what people need, and this feature is created now for contact tracing, but this goes back to the question if we should allow the user to choose what fields they need.

If this is going to be used for things other than contact tracing specifically (which I suspect it will), I think it makes sense to allow the user to only select the fields they need.

For contact tracing, venue owners require email, and sometimes the event will require users to enter their phone number in the feedback field.

Reply:
We cannot send the `feedback` field to the venue owner without proper data cleaning, and I am 110% that xCOM don't care about doing that. The feedback field can potentially contain other, maybe "sensitive" stuff that isn't meant for for others than the host. eg. 5-10% of people write their preference of food there, and that is _not_ relevant for the venue owner at all.

Reply:
Also, if xCOM starts giving such a list to the venue owner, they *must* at least state that in the event description, and give a clear and simple desc about how the data is used, by who, where, for what purpose, and for how long it is used before being deleted, so that the user knows it in advance, and can skip registering in case he don't want to attend.

Reply:
I agree, but again I think this need to be discussed a bit further.

About the disclosure, the frontend part of this does include a message when downloading stating exactly that. @erlingfn will create a PR soon

Reply:
What `disclosure`, stating what? Not sure if we are talking about the same stuff. Do we really think xCOM will manually filter these lists before throwing them at someone?

Reply:
A disclosure stating that all attendees must be informed in advance about what data will be shared and who it will be shared with. Just so that event admins are aware of the rules here. I.e the user needs to confirm that he/she understands this before exporting the list.

Reply:
That is probably a bit late when they are ready to export the list tho. :shrug:

Reply:
I don't really see any other way to do it? If they don't know that they need to share the data before registrations open, there's really nothing we can do (i.e. they can't share the data).

We could have something like a field on the event informing the user that the data will be shared, but I think that just creates more problems. Since then I think admins won't be as careful with stating who it will be shared with etc.